webapp/src/main/java/com/box/l10n/mojito/quartz/QuartzPollableTaskScheduler.java

```diff
@@ -88,6 +90,15 @@ public <I, O> PollableFuture<O> scheduleJob(QuartzJobInfo<I, O> quartzJobInfo) {
 
     Scheduler scheduler = schedulerManager.getScheduler(quartzJobInfo.getScheduler());
 
+    if (quartzJobInfo.getClazz().isAnnotationPresent(DisallowConcurrentExecution.class)
```

Comment:
Interesting. What's the rational behind checking for`DisallowConcurrentExecution`? Is it just to avoid checking too often? I'm not sure it truly makes sense to have limit to those types of job vs any jobs with a provided unique id. Practically, I think it will apply only to job that `DisallowConcurrentExecution` though. 

At first, I was thinking it is not needed and we could just be inferred from the older check. 

```
if (!scheduler.checkExists(triggerKey)) {
        logger.debug("Schedule QuartzPollableJob with key: {}", keyName);
        scheduler.scheduleJob(jobDetail, trigger);
      } else {
        logger.debug("Job already scheduled for key: {}, reschedule", keyName);
        scheduler.rescheduleJob(triggerKey, trigger);
      }
      ``` 

The problem, I now see is that it builds the "uniqueId" from the PollableTask in case it is a one off jobs and it prevents moving the task recreation to later. 

One alternative, would be to just mark the created PollableTask as "skipped". It would simplify the code, have tracking of how many attempts were made/skipped at the cost of some noise in the database. 

Just throwing more idea here. We could stop using the pollableTaksId to build the unique ID and just use a UUID. Then we loose the direct link pollable task to Quartz job though via the shared id. Can be nice for debugging but not sure we use it that much.

Reply:
I used DisallowConcurrentExecution to limit the instances where this type of logic will execute to only jobs where concurrency is disabled, primarily because that aligns with the RepositoryStatisticsJob config. 

My initial approach was similar to what you've suggested, find the existing pollable, mark it as finished with a message override indicating it was skipped but then changed it as I figured we could avoid the additional entity creation and updating cost by returning the existing pollable task.